crates/bevy_gizmos/src/frustum.rs

```diff
@@ -0,0 +1,213 @@
+    ///
+    /// Defaults to `None`.
+    pub default_color: Option<Color>,
+}
+
+/// Add this [`Component`] to an entity to draw its [`Frustum`] component.
+#[derive(Component, Reflect, Default, Debug)]
+#[reflect(Component, FromReflect, Default)]
+pub struct FrustumGizmo {
+    /// The color of the frustum.
+    ///
+    /// The default color from the [`GizmoConfig`] resource is used if `None`,
+    pub color: Option<Color>,
+}
+
+fn frustum_gizmos(
+    query: Query<
+        (Entity, &Frustum, &FrustumGizmo, Option<&Handle<LineGizmo>>),
+        Or<(Changed<Frustum>, Changed<FrustumGizmo>)>,
+    >,
+    config: Res<GizmoConfig>,
+    mut commands: Commands,
+    mut lines: ResMut<Assets<LineGizmo>>,
+    mut removed: RemovedComponents<FrustumGizmo>,
+) {
+    for entity in removed.read() {
+        if !query.contains(entity) {
+            commands.entity(entity).remove::<Handle<LineGizmo>>();
+        }
+    }
+
+    for (entity, frustum, gizmo, line_handle) in &query {
+        let color = gizmo
+            .color
+            .or(config.frustum.default_color)
+            .unwrap_or_else(|| color_from_entity(entity));
+
+        frustum_inner(
+            &mut commands,
+            &mut lines,
+            entity,
+            frustum,
+            line_handle,
+            color,
+        );
+    }
+}
+
+fn all_frustum_gizmos(
+    query: Query<
+        (Entity, &Frustum, Option<&Handle<LineGizmo>>),
+        (Without<FrustumGizmo>, Changed<Frustum>),
+    >,
+    config: Res<GizmoConfig>,
+    mut commands: Commands,
+    mut lines: ResMut<Assets<LineGizmo>>,
+) {
+    for (entity, frustum, line_handle) in &query {
+        let color = config
+            .frustum
+            .default_color
+            .unwrap_or_else(|| color_from_entity(entity));
+
+        frustum_inner(
+            &mut commands,
+            &mut lines,
+            entity,
+            frustum,
+            line_handle,
+            color,
+        );
+    }
+}
+
+fn frustum_inner(
+    commands: &mut Commands,
+    lines: &mut ResMut<Assets<LineGizmo>>,
+    entity: Entity,
+    frustum: &Frustum,
+    line_handle: Option<&Handle<LineGizmo>>,
+    color: Color,
+) {
+    let Some(positions) = calculate_frustum_positions(frustum) else {
+        return;
+    };
+
+    let line = LineGizmo {
+        colors: std::iter::repeat(color.as_linear_rgba_f32())
+            .take(positions.len())
+            .collect(),
+        positions,
+        strip: true,
+    };
+
+    if let Some(handle) = line_handle {
+        lines.insert(handle, line);
+    } else {
+        commands.entity(entity).insert(lines.add(line));
+    }
+}
+
+fn remove_frustum_gizmos(
+    query: Query<Entity, (With<Handle<LineGizmo>>, Without<FrustumGizmo>)>,
+    mut commands: Commands,
+) {
+    for entity in &query {
+        commands.entity(entity).remove::<Handle<LineGizmo>>();
+    }
+}
+
+fn calculate_frustum_positions(frustum: &Frustum) -> Option<Vec<[f32; 3]>> {
```

Comment:
This and the below should probably be implemented on `Frustum` as someone might want it sometime. :)